wallet/txauthor/author.go

```diff
@@ -109,6 +109,10 @@ func NewUnsignedTransaction(outputs []*wire.TxOut, relayFeePerKb dcrutil.Amount,
 		scriptSizes = append(scriptSizes, inputDetail.RedeemScriptSizes...)
 
 		maxSignedSize = txsizes.EstimateSerializeSize(scriptSizes, outputs, changeScriptSize)
+		if maxSignedSize > maxTxSize {
+			continue
```

Comment:
If i'm reading this correctly, this will put the algorithm in an infinite loop. It will continue to fetch inputs for the exact same target value and keep hitting this condition.